getting_started/build-guide/build-with-zephyr.rst

```diff
@@ -12,108 +12,217 @@ This guide describes how to build and run |SOF| as a Zephyr application.
-     git fetch sof
-     git switch --track sof/main
+      west update
 
-     # If needed, correct submodule URLs in .git/config with latest .gitmodules file.
-     git submodule sync --recursive
-     # Download submodules
-     git submodule update --init --recursive
+   .. warning::
+      This command will check out revisions specified in ``$ZEPHYR_WORKSPACE/sof/west.yml`` file for projects:
+        - Rimage (SOF submodule)
+        - Tomlc99 (Rimage submodule)
+        - Zephyr
+        - projects in ``$ZEPHYR_WORKSPACE/modules`` directory.
 
-   Alternatively, feel free to delete the ``sof`` clone downloaded by
-   ``west`` and replace it with any other ``sof`` clone you already
-   have; west will automatically adjust.
+      Make sure to backup your work before changing revisions!
+      It will not affect your SOF project revision.
 
-#. Build and sign a firmware image:
+#. Build a board - make sure the appropriate Zephyr SDK or other toolchain of your
+   choice. Boards to build are listed in ``$ZEPHYR_WORKSPACE/sof/app/boards`` directory.
 
    .. code-block:: bash
 
-      cd $ZEPHYR_WORKSPACE
-      ./modules/audio/sof/scripts/xtensa-build-zephyr.sh -h  # shows usage
-      ./modules/audio/sof/scripts/xtensa-build-zephyr.sh $your_platforms
-      ls build-*/zephyr/zephyr.*
-        => build-*/zephyr/zephyr.ri ...
+      west build --build-dir build-tgl --board intel_adsp_cavs25 ./sof/app
 
+   .. hint::
+      SOF project defines platform names that have Zephyr board counterpart.
+      In this example platform *Tigerlake* matches Zephyr board *inteL_adsp_cavs25*
+      (this is why output directory is named *build-tgl* however you may use any name you wish).
 
-There is work in progress to substitute ``xtensa-build-zephyr.sh`` with
-a Python replacement ``xtensa-build-zephyr.py`` for Windows
-compatibility. It can already be used in many cases.
+   .. tip::
+      - To add verbosity to the build output use -v -v flags. Example:
+        ``west -v -v build --build-dir build-tgl --board intel_adsp_cavs25 ./sof/app``
 
-Run
-***
+      - To perform complete clean rebuild use --pristine flag. Example:
+        ``west -v -v build --build-dir build-tgl --pristine always --board intel_adsp_cavs25 ./sof/app``
+
+   ``.elf`` file produced by ``west build`` is missing a
+   manifest and signature. You need to sign the file using **rimage tool**
+   and a **private key** to generate final firmware image (``.ri`` file).
+
+#. Build rimage tool
 
-For convenience, the ``xtensa-build-zephyr.sh`` script copies all
-firmware files into a single, "staging" directory:
+   .. code-block:: bash
+
+      cmake -B ./build-rimage -S ./sof/rimage
+      cmake --build ./build-rimage
+
+#. Sign firmware using rimage tool and a private key
 
    .. code-block:: bash
 
-      $ tree build-sof-staging/
+      west sign --build-dir ./build-tgl -t rimage --tool-path ./build-rimage/rimage --tool-data ./sof/rimage/config -- -k ./sof/keys/otc_private_key_3k.pem
```

Comment:
do we want to mention the -f option? or the version in manifest will be 0.0.0, which is not correct.

Reply:
-f option is for `-f, --force           ignore any errors and try to proceed` according to `west sign --help` .
How is that related to versioning?